automation/download_datafusion_proto.py

```
@@ -0,0 +1,100 @@
+    """Extract import paths from a .proto file."""
+    return re.findall(r'^import\s+"([^"]+)"\s*;', proto_path.read_text(), re.MULTILINE)
+
+
+def main():
+    version = get_datafusion_version()
+    print(f"Downloading datafusion proto files for version {version}")
+
+    # Download main proto file into PROTO_DIR root
+    main_url = f"{BASE_URL}{version}/{MAIN_PROTO}"
+    main_output = PROTO_DIR / "datafusion.proto"
+    print(f"  datafusion.proto")
```

Ruff (0.15.6)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)